kratos/mpi/utilities/gather_modelpart_utility.h

```diff
@@ -176,6 +241,20 @@ class KRATOS_API(KRATOS_MPI_CORE) GatherModelPartUtility
     ///@name Private Operations
     ///@{
 
+    /**
+     * @brief Function to gather entities from other partitions
+     * @details This function is intended to gather entities from other partitions. The map indicates the partitions to gather and the vector the entities to gather from each partition. In the current serial case it does nothing.
+     * @param rModelPart Model part to gather entities from other partitions
+     * @param rEntitiesToBring Entities to gather from other partitions
+     * @param EchoLevel The verbosity level
+     */
+    template <class TObjectType>
+    static void GatherEntityFromOtherPartitions(
+        ModelPart& rModelPart,
+        const std::map<int, std::vector<std::size_t>>& rEntitiesToBring,
```

Comment:
why map and not unordered map?

Also is it the ids if the entities to bring? The template seems unused, maybe it should be `std::vector<TObjectType>` insetad?

Reply:
Huuum, I need to be ordered to avoid conflicts in the communications

Reply:
what exactly does need to be ordered?
I mean one could still pass the entities in a different order on different ranks no?

> Also is it the ids if the entities to bring? The template seems unused, maybe it should be std::vector<TObjectType> insetad?

and what about this?

Reply:
It requires to be ordered, otherwise doesn't work. I need to implement the asynchronous. I will try bring to the data communicator your implementation in the MappingApplication.

Reply:
BTW, the template is used, look at thde code. There are constexpr

Reply:
ok

can you then please update the function descriptions to include this info?
I.e. what those maps represent and what restrictions there are

Reply:
Done